### ISSUE TITLE
Build failure: tuxedo-keyboard

### ISSUE THREAD

**Issue by chronicc** [NixOS/nixpkgs#241423]
### Steps To Reproduce

1. build *X*

### Build log

```
error: builder for '/nix/store/n3rz4cm35k6bg2y0980makqbw2n6zirh-tuxedo-keyboard-6.4-3.1.4.drv' failed with exit code 2;
       last 10 log lines:
       >   230 | struct class * __must_check class_create(const char *name);
       >       |                             ^~~~~~~~~~~~
       > /build/source/./src/tuxedo_io/tuxedo_io.c: At top level:
       > /build/source/./src/tuxedo_io/tuxedo_io.c:55:13: warning: 'dmi_string_in' defined but not used [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-function-Wunused-function8;;]
       >    55 | static bool dmi_string_in(enum dmi_field f, const char *str)
       >       |             ^~~~~~~~~~~~~
       > cc1: some warnings being treated as errors
       > make[2]: *** [/nix/store/z3zrcbgzhp8x86gg83zwvlxc6prwkinb-linux-6.4-dev/lib/modules/6.4.0/source/scripts/Makefile.build:252: /build/source/./src/tuxedo_io/tuxedo_io.o] Error 1
       > make[1]: *** [/nix/store/z3zrcbgzhp8x86gg83zwvlxc6prwkinb-linux-6.4-dev/lib/modules/6.4.0/source/Makefile:2026: /build/source] Error 2
       > make: *** [Makefile:29: all] Error 2
       For full logs, run 'nix log /nix/store/n3rz4cm35k6bg2y0980makqbw2n6zirh-tuxedo-keyboard-6.4-3.1.4.drv'.
error: 1 dependencies of derivation '/nix/store/z54k5v3fqik5mphv3kq2bcylzsf1198q-kernel-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hb81nbbxfayyxwgg93cd3q04qc64gkgx-linux-6.4-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lf2gn8j129zsxiy9r8g54x7n7q3k34wk-nixos-system-libre-23.05.20230701.0de8605.drv' failed to build
```

`nix log` output: https://gist.github.com/chronicc/cba77cdb19fb56ff7c172ac853ad82c8

### Additional context

Happened after a `nix flak update`. Using the 23.05 release of nixpkgs.

```
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/b6c73c5fe53bb3afbf65e870541e0645e9145171' (2023-06-20)
  → 'github:nixos/nixpkgs/0de86059128947b2438995450f2c2ca08cc783d5' (2023-07-01)
```

### Notify maintainers

@blanky0230 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.3.8, NixOS, 23.05 (Stoat), 23.05.20230620.b6c73c5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(chronicc): `""`
 - channels(root): `"nixos-23.05, nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by chronicc** [comment]
As a temporary workaround I am using the 6.3 kernel with

```nix
boot.kernelPackages = pkgs.linuxPackages_6_3;
```

**Comment by Occhioverde** [comment]
As of Nixpkgs 8acef30 (current unstable), which includes tuxedo-keyboard 3.2.7, this issue seems to be solved (as the build problem was fixed in tuxedo-keyboard 3.2.6, see https://github.com/tuxedocomputers/tuxedo-keyboard/issues/183#issuecomment-1617047673).

**Comment by blanky0230** [comment]
@chronicc As I see you have found a workaround for now. As @Occhioverde stated the issue has been fixed wih tuxedo-keyboard 3.2.7 Is this info sufficient for you for now?

**Comment by chronicc** [comment]
@blanky0230 Yes, the build with latest kernel package works again. Thanks!
